rollup/sync_service/sync_service.go

```diff
@@ -237,6 +237,14 @@ func (s *SyncService) fetchMessages() {
 			return
 		}
 
+		// write start index of very first L1MessageV2 to database
+		if len(msgsV2) > 0 && rawdb.ReadL1MessageV2StartIndex(s.db) == nil {
```

Comment:
Since writes are batched in `batchWriter` but here you're reading from `s.db`, I'm worried that the 2nd, etc. iterations will overwrite the correct index here.

Seems more correct to use `rawdb.ReadL1MessageV2StartIndex(batchWriter)` (need to double check how reads from a batch work), or just store in a variable and write to db in `flush`.

Reply:
Introduced a variable. This way we also do less DB reads.